src/org/mozilla/app/push/util/SQLTransaction.kt

```diff
@@ -0,0 +1,25 @@
+package org.mozilla.app.push.util
+
+import java.sql.Connection
+import java.sql.SQLException
+
+fun <T> Connection.transaction(logging: String = "SQLTransaction", action: () -> T): T? {
+    return try {
+        this.use {
+
+            this.autoCommit = false
+
+            val result = action()
+
+            this.commit()
+
+            this.autoCommit = true
+
+            return@use result
+        }
+    } catch (e: SQLException) {
+        e.printStackTrace()
+        logger().error("[worker][$logging][error]$e")
+        return null
```

Comment:
1. exception was already printed in log. Maybe you can remove this line ` e.printStackTrace()`?
2. Might think over about `return null` or throwing a runtime exception.

Reply:
Can I just rethrow the exception here?